graylog-storage-opensearch2/src/test/java/org/graylog/storage/opensearch2/testing/DatanodeInstance.java

```diff
@@ -0,0 +1,152 @@
+import org.testcontainers.images.ImagePullPolicy;
+import org.testcontainers.images.PullPolicy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.net.URI;
+import java.util.Locale;
+
+import static java.util.Objects.isNull;
+
+public class DatanodeInstance extends TestableSearchServerInstance {
+    private static final Logger LOG = LoggerFactory.getLogger(DatanodeInstance.class);
+
+    public static final String DEFAULT_HEAP_SIZE = "2g";
+    public static final SearchServer DATANODE_VERSION = SearchServer.DATANODE_DEV;
+
+    private final OpenSearchClient openSearchClient;
+    private final Client client;
+    private final FixtureImporter fixtureImporter;
+    private final Adapters adapters;
+
+    protected DatanodeInstance(String image, SearchVersion version, Network network, String heapSize) {
+        super(image, version, network, heapSize);
+        RestHighLevelClient restHighLevelClient = buildRestClient();
+        this.openSearchClient = new OpenSearchClient(restHighLevelClient, false, new ObjectMapperProvider().get());
+        this.client = new ClientOS2(this.openSearchClient);
+        this.fixtureImporter = new FixtureImporterOS2(this.openSearchClient);
+        adapters = new AdaptersOS2(openSearchClient);
+    }
+    protected DatanodeInstance(String image, SearchVersion version, Network network) {
+        this(image, version, network, DEFAULT_HEAP_SIZE);
+    }
+
+    @Override
+    public SearchServer searchServer() {
+        return DATANODE_VERSION;
+    }
+
+    private RestHighLevelClient buildRestClient() {
+        return new RestHighLevelClientProvider(
+                new GracefulShutdownService(),
+                ImmutableList.of(URI.create("http://" + this.getHttpHostAddress())),
+                Duration.seconds(60),
+                Duration.seconds(60),
+                Duration.seconds(60),
+                1,
+                1,
+                1,
+                false,
+                null,
+                Duration.seconds(60),
+                "http",
+                false,
+                false,
+                new BasicCredentialsProvider())
+                .get();
+    }
+
+    // Caution, do not change this signature. It's required by our container matrix tests. See SearchServerInstanceFactoryByVersion
+    public static DatanodeInstance create(SearchVersion searchVersion, Network network) {
+        return create(searchVersion, network, DEFAULT_HEAP_SIZE);
+    }
+
+    private static DatanodeInstance create(SearchVersion searchVersion, Network network, String heapSize) {
+        final String image = imageNameFrom(searchVersion.version());
+
+        LOG.debug("Creating instance {}", image);
+
+        return new DatanodeInstance(image, searchVersion, network, heapSize);
+    }
+
+    protected static String imageNameFrom(Version version) {
+        return String.format(Locale.ROOT, "graylog/graylog-datanode:%s", "dev"); // TODO:use the requested version
```

Comment:
is the TODO still a TODO?

Reply:
It is. The only tag we [publish](https://hub.docker.com/r/graylog/graylog-datanode/tags) now in the `dev`. So we can't request an image with specific version.